backend/src/main/java/com/festago/festival/repository/FestivalSpecification.java

```diff
@@ -0,0 +1,34 @@
+package com.festago.festival.repository;
+
+import com.festago.festival.domain.Festival;
+import java.time.LocalDate;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class FestivalSpecification {
+
+    private static final String START_DATE = "startDate";
+    private static final String END_DATE = "endDate";
+
+    public static Specification<Festival> all() {
+        return (root, query, criteriaBuilder) -> criteriaBuilder.conjunction();
+    }
+
+    public static Specification<Festival> afterStartDate(LocalDate currentTime) {
+        return (root, query, criteriaBuilder) -> criteriaBuilder.lessThan(root.get(START_DATE), currentTime);
+    }
+
+    public static Specification<Festival> beforeStartDate(LocalDate currentTime) {
+        return (root, query, criteriaBuilder) -> criteriaBuilder.greaterThan(root.get(START_DATE), currentTime);
+    }
+
+    public static Specification<Festival> afterEndDate(LocalDate currentTime) {
+        return (root, query, criteriaBuilder) -> criteriaBuilder.lessThan(root.get(END_DATE), currentTime);
+    }
+
+    public static Specification<Festival> beforeEndDate(LocalDate currentTime) {
+        return (root, query, criteriaBuilder) -> criteriaBuilder.greaterThan(root.get(END_DATE), currentTime);
+    }
```

Comment:
위의 커멘트에도 남기긴 했지만...
다음과 같이 메서드가 하는 일을 명확하게 하는것은 어떤가요
```suggestion
    public static Specification<Festival> all(LocalDate currentTime) { // 메서드 참조를 위해 파라미터 추가
        return (root, query, criteriaBuilder) -> criteriaBuilder.conjunction();
    }

    public static Specification<Festival> progress(LocalDate currentTime) {
        return (root, query, criteriaBuilder) -> criteriaBuilder.and(
            criteriaBuilder.lessThan(root.get(START_DATE), currentTime), // lessThanOrEqualTo 고려할 것
            criteriaBuilder.greaterThan(root.get(END_DATE), currentTime) // greaterThanOrEqualTo 고려할 것
        );
    }

    public static Specification<Festival> beforeStartDate(LocalDate currentTime) {
        return (root, query, criteriaBuilder) -> criteriaBuilder.greaterThan(root.get(START_DATE), currentTime);
    }

    public static Specification<Festival> afterEndDate(LocalDate currentTime) {
        return (root, query, criteriaBuilder) -> criteriaBuilder.lessThan(root.get(END_DATE), currentTime);
    }
```